File: app/src/main/java/com/friendlypos/principal/helpers/SubirHelperProductoDevuelto.java
package com.friendlypos.principal.helpers;

import android.app.Activity;
import android.content.Context;
import android.util.Log;
import android.widget.Toast;

import com.friendlypos.app.broadcastreceiver.NetworkStateChangeReceiver;
import com.friendlypos.application.datamanager.BaseManager;
import com.friendlypos.application.interfaces.RequestInterface;
import com.friendlypos.distribucion.modelo.Inventario;
import com.friendlypos.distribucion.modelo.Pivot;
import com.friendlypos.login.util.SessionPrefes;
import com.friendlypos.preventas.modelo.EnviarClienteVisitado;
import com.friendlypos.preventas.modelo.visit;
import com.friendlypos.principal.activity.MenuPrincipal;
import com.friendlypos.principal.modelo.EnviarProductoDevuelto;

import retrofit2.Call;
import retrofit2.Callback;
import retrofit2.Response;

import static android.content.ContentValues.TAG;

/**
 * Created by Delvo on 02/12/2017.
 */

public class SubirHelperProductoDevuelto {

    private NetworkStateChangeReceiver networkStateChangeReceiver;
    private MenuPrincipal activity;
    private Context mContext;
    private RequestInterface mAPIService;
    int codigoServer;
    int codigo;
    String codigoS;
    String mensajeS;
    String resultS;
    public SubirHelperProductoDevuelto(MenuPrincipal activity) {
        this.activity = activity;
        this.mContext = activity;
        networkStateChangeReceiver = new NetworkStateChangeReceiver();
        mAPIService = BaseManager.getApi();
    }

    public void sendPostClienteProductoDevuelto(EnviarProductoDevuelto facturaQuery, final int cantidadFactura) {
        String token = "Bearer " + SessionPrefes.get(mContext).getToken();
        Log.d("tokenCliente", token + " ");
        if (isOnline()) {
            Log.d("factura1", facturaQuery + " ");
            mAPIService.savePostProductoDevuelto(facturaQuery, token).enqueue(new Callback<Inventario>() {

                @Override
                public void onResponse(Call<Inventario> call, Response<Inventario> response) {


                    if(response.isSuccessful()) {
                        Log.d("respPreventa",response.body().toString());

                        codigo = response.code();
                        codigoS = response.body().getCode();
                        mensajeS = response.body().getMessages();
                        resultS= String.valueOf(response.body().isResult());

                        activity.codigoDeRespuestaProductoDevuelto(codigoS, mensajeS, resultS, codigo, cantidadFactura);
                        Log.d("codID",cantidadFactura+"");

                    }
                    else{
                        Log.d("respPreventa", "Error");
                    }

                }

                @Override
                public void onFailure(Call<Inventario> call, Throwable t) {
                    Log.e(TAG, "Unable to submit post to API.");
                }
            });}
        else{
            Toast.makeText(activity, "Error, por favor revisar conexión de Internet", Toast.LENGTH_SHORT).show();
        }
    }

    private boolean isOnline() {
        return networkStateChangeReceiver.isNetworkAvailable(mContext);
    }
}

File: app/src/main/java/com/friendlypos/principal/helpers/SubirHelperProductoDevuelto.kt
package com.friendlysystemgroup.friendlypos.principal.helpers

import android.content.ContentValues
import android.content.Context
import android.util.Log
import android.widget.Toast
import com.friendlysystemgroup.friendlypos.app.broadcastreceiver.NetworkStateChangeReceiver
import com.friendlysystemgroup.friendlypos.application.datamanager.BaseManager.Companion.api
import com.friendlysystemgroup.friendlypos.application.interfaces.RequestInterface
import com.friendlysystemgroup.friendlypos.distribucion.modelo.Inventario
import com.friendlysystemgroup.friendlypos.distribucion.modelo.Inventario.isResult
import com.friendlysystemgroup.friendlypos.distribucion.modelo.Inventario.toString
import com.friendlysystemgroup.friendlypos.login.util.SessionPrefes.Companion.get
import com.friendlysystemgroup.friendlypos.principal.activity.MenuPrincipal
import com.friendlysystemgroup.friendlypos.principal.modelo.EnviarProductoDevuelto

/**
 * Created by Delvo on 02/12/2017.
 */
class SubirHelperProductoDevuelto(private val activity: MenuPrincipal) {
    private val networkStateChangeReceiver =
        NetworkStateChangeReceiver()

    private val mContext: Context = activity
    private val mAPIService: RequestInterface? = api
    var codigoServer: Int = 0
    var codigo: Int = 0
    var codigoS: String? = null
    var mensajeS: String? = null
    var resultS: String? = null

    fun sendPostClienteProductoDevuelto(
        facturaQuery: EnviarProductoDevuelto,
        cantidadFactura: Int
    ) {
        val token = "Bearer " + get(mContext).token
        Log.d("tokenCliente", "$token ")
        if (isOnline) {
            Log.d("factura1", "$facturaQuery ")
            mAPIService!!.savePostProductoDevuelto(facturaQuery, token)
                .enqueue(object : retrofit2.Callback<Inventario?> {
                    override fun onResponse(
                        call: retrofit2.Call<Inventario?>?,
                        response: retrofit2.Response<Inventario?>
                    ) {
                        if (response.isSuccessful()) {
                            Log.d("respPreventa", response.body().toString())

                            codigo = response.code()
                            codigoS = response.body().code
                            mensajeS = response.body().messages
                            resultS = response.body().isResult.toString()

                            activity.codigoDeRespuestaProductoDevuelto(
                                codigoS!!, mensajeS,
                                resultS!!, codigo, cantidadFactura
                            )
                            Log.d("codID", cantidadFactura.toString() + "")
                        } else {
                            Log.d("respPreventa", "Error")
                        }
                    }

                    override fun onFailure(call: retrofit2.Call<Inventario?>?, t: Throwable?) {
                        Log.e(ContentValues.TAG, "Unable to submit post to API.")
                    }
                })
        } else {
            Toast.makeText(
                activity,
                "Error, por favor revisar conexión de Internet",
                Toast.LENGTH_SHORT
            ).show()
        }
    }

    private val isOnline: Boolean
        get() = networkStateChangeReceiver.isNetworkAvailable(mContext)
}
